### ISSUE TITLE
racket-repl: Wrong type argument: window-live-p, nil

### ISSUE THREAD

**Issue by xuchunyang** [greghendershott/racket-mode#411]
1. Launch the Racket REPL with C-c C-c from a racket buffer with the following

```racket
#lang racket
(+ 1 2)
```

2. Delete the REPL window with C-x 0

3. Try to switch to the REPL from source with C-c C-z (`racket-repl`), the following backtrace occurs

> Debugger entered--Lisp error: (wrong-type-argument window-live-p nil)
>   (select-window nil)
>   (racket-repl nil)
>   (funcall-interactively racket-repl nil)
>   (call-interactively racket-repl nil nil)
>   (command-execute racket-repl)
>

**Comment by greghendershott** [comment]
I'm not able to reproduce this. In fact I do something like this frequently when using Racket Mode myself.

However I'll look at the code and try to think about how it might occur given some conditions you have  but I do not.

**Comment by xuchunyang** [comment]
Hmm, that's odd, https://github.com/greghendershott/racket-mode/blob/master/racket-repl.el#L156

```el
(select-window (get-buffer-window racket--repl-buffer-name t))
```

when there is no window showing the racket repl buffer, `get-buffer-window` returns nil, then the error occurs.

**Comment by greghendershott** [comment]
That makes sense. Indeed when I do `M-x eval` and enter `(get-buffer-window racket--repl-buffer-name t)` it says `nil`.

And yet!!

1. When I do `M-x racket-repl` or <kbd>C-c C-z</kbd>, it works -- activates the REPL buffer in a window.

2. When I <kbd>C-M-x</kbd> the definition of `racket-repl` in `racket-repl.el`, to instrument it for debugging, then <kbd>C-c C-z</kbd>, then step through the expressions, it works. On the expression `(get-buffer-window racket--repl-buffer-name t)` it says `Result: #<window 125 on *Racket REPL*>`. What is this window? How is it auto-magically created? And why for me but not you?

---

TL;DR: The code doesn't make sense and I will fix it. At the same time, I am super curious to learn the mystery of how/why it is sometimes works!

**Comment by greghendershott** [comment]
I looked at this again with a fresher brain. It's not so mysterious why the code should work. The full code (omitting doc string) is:

```elisp
(defun racket-repl (&optional noselect)
  (interactive "P")
  (if (racket--repl-live-p)
      (racket--repl-show-and-move-to-end)
    (racket--repl-start))
  (unless noselect
    (select-window (get-buffer-window racket--repl-buffer-name t))))
```

Both arms of the `if` expression -- both `(racket--repl-show-and-move-to-end)` and `(racket--repl-start)` -- display the buffer in a window. Both call `display-buffer`.

[Maybe I should add a comment to the code, pointing this out. It would have helped remind me what I did. :smile:]

The only work left to do, is optionally select the window (in modern UI terms, "give it the focus") that already exists.

So the mystery is, how/why on your system, is the buffer not being displayed in a window, prior to you hitting this error.

---

Trying to think how to narrow this down:

1. Do you experience this when the `*Racket REPL*` buffer (a) already exists, (b) does not already exist, or (c) both?

2. May I see more about your configuration? Can you paste the output of `M-x racket-bug-report` into a comment here?

**Comment by xuchunyang** [comment]
Oops, totally my bad, I had the following in my init file for some reason that I can't remember

    (advice-add 'racket--repl-show-and-move-to-end :override #'ignore)
